### PR TITLE
Add Product Span Positioning

### PR DESCRIPTION
**Pull request by czapplitec**
To address #251 and #227

### CODE REVIEW

File: IFC4x3/Templates/Object Connectivity/Product Span Positioning/Documentation.md [hunk 1]
```diff
@@ -0,0 +1 @@
+An _IfcProduct_ can be placed relative to two _IfcReferent_'s, which indicate the start and end positions of the product along a linear section.
```
Comment (larswik) [inline IFC4x3/Templates/Object Connectivity/Product Span Positioning/Documentation.md:1]:
```suggestion
An _IfcProduct_ can be placed relative to two _IfcReferent_s, which indicate the start and end positions of the product along a linear section.
```

**Review by larswik — CHANGES_REQUESTED** [review]

File: IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocTemplateDefinition xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Product Span Positioning" UniqueId="1cdcaa9f-85d5-42cf-9aaf-0fd663b14ee8" Type="IfcProduct">
+	<Rules>
+		<DocModelRuleAttribute Name="PositionedRelativeTo">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml:4]: does ther eneed to be some sort of Implied ordering in the IfcReferents?

or will we address this using GU and name labels

Comment (larswik) [reply IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml:4]: According to the discussion between me, @czapplitec and @SergejMuhic was that this was a GU issue and to use the Name attribute of IfcRelPositions to capture the ordering (start/end) - hence the rule ids for the names.

Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml:4]: Awseome, i will put it on the GU work/issue list.

**Review by AlexBrad1eyCT — APPROVED** [review]

File: IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+						<DocModelRuleAttribute Name="Name" Identification="StartPosition">
+							<Rules>
+								<DocModelRuleEntity Name="IfcLabel" />
+							</Rules>
+						</DocModelRuleAttribute>
+					</Rules>
+				</DocModelRuleEntity>
+			</Rules>
+		</DocModelRuleAttribute>
+		<DocModelRuleAttribute Name="PositionedRelativeTo">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml:30]:
```suggestion
		<DocModelRuleAttribute Name="PositionedRelativeTo" Identification="EndPositionedRelativeTo">
```
```suggestion
		<DocModelRuleAttribute Name="PositionedRelativeTo">
```

File: IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+										<DocModelRuleAttribute Name="PredefinedType" Identification="EndPositionType">
+											<Rules>
+												<DocModelRuleEntity Name="IfcReferentTypeEnum" />
+											</Rules>
+										</DocModelRuleAttribute>
+									</Rules>
+								</DocModelRuleEntity>
+							</Rules>
+						</DocModelRuleAttribute>
+						<DocModelRuleAttribute Name="Name" Identification="EndPosition">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml:47]:
```suggestion
						<DocModelRuleAttribute Name="Name" Identification="EndPositionName">
```

File: IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+										<DocModelRuleAttribute Name="PredefinedType" Identification="StartPositionType">
+											<Rules>
+												<DocModelRuleEntity Name="IfcReferentTypeEnum" />
+											</Rules>
+										</DocModelRuleAttribute>
+									</Rules>
+								</DocModelRuleEntity>
+							</Rules>
+						</DocModelRuleAttribute>
+						<DocModelRuleAttribute Name="Name" Identification="StartPosition">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Templates/Object Connectivity/Product Span Positioning/DocTemplateDefinition.xml:21]:
```suggestion
						<DocModelRuleAttribute Name="Name" Identification="StartPositionName">
```

**Review by AlexBrad1eyCT — APPROVED** [review]
Just some name updates

**Comment by AlexBrad1eyCT** [comment]
Merging due to deadline, any corrects submit via new bug fix

**Pull request merged**